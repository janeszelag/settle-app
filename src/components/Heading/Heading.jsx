import React from "react"
import styled from "styled-components/native"
import PropTypes from "prop-types"
import { gunmetal } from "../../styles/colors"
import { useWindowDimensions, Platform } from 'react-native'


const Heading = ({ children, level }) => {
  const windowWidth = useWindowDimensions().width
  const getFontSize = ({ level }) => {
    let fontSize
    switch (level) {

      case "h1":
        fontSize = (Platform.OS === 'web' && windowWidth <= 764) ? 28 : 44
        break
      case "h2":
        fontSize = (Platform.OS === 'web' && windowWidth <= 764) ? 24 : 28
        break
      case "h3":
        fontSize = 20
        break
      case "h4":
        fontSize = 16
        break
    }
    return {
      fontSize
    }
  }

  const StyledText = styled.Text(
    getFontSize,
    {
      fontWeight: "bold",
      color: gunmetal,
    })

  return <StyledText level={level}>{children}</StyledText>
}


Heading.propTypes = {
  /**
   * The visual level of the heading.
   */
  level: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']).isRequired,
  /**
   * The content. Can be text, other components.
   */
  children: PropTypes.string.isRequired,
}


export default Heading
