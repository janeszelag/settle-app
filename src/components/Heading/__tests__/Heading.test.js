import React from "react"
import renderer from "react-test-renderer"
import Heading from "../Heading"

describe("<Heading />", () => {
  it("has 1 child", () => {
    const tree = renderer
      .create(<Heading level='h1'>Heading</Heading>)
      .toJSON()
    expect(tree.children.length).toBe(1)
  })

  it("renders an h1 level heading", () => {
    const tree = renderer
      .create(<Heading level='h1'>Heading h1</Heading>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders an h2 level heading", () => {
    const tree = renderer
      .create(<Heading level='h2'>Heading h2</Heading>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders an h3 level heading", () => {
    const tree = renderer
      .create(<Heading level='h3'>Heading h3</Heading>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders an h4 level heading", () => {
    const tree = renderer
      .create(<Heading level='h4'>Heading h4</Heading>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
