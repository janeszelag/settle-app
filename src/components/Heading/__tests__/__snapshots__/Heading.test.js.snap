// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders an h1 level heading 1`] = `
<Text
  level="h1"
  style={
    Array [
      Object {
        "color": "#293241",
        "fontSize": 44,
        "fontWeight": "bold",
      },
    ]
  }
>
  Heading h1
</Text>
`;

exports[`<Heading /> renders an h2 level heading 1`] = `
<Text
  level="h2"
  style={
    Array [
      Object {
        "color": "#293241",
        "fontSize": 28,
        "fontWeight": "bold",
      },
    ]
  }
>
  Heading h2
</Text>
`;

exports[`<Heading /> renders an h3 level heading 1`] = `
<Text
  level="h3"
  style={
    Array [
      Object {
        "color": "#293241",
        "fontSize": 20,
        "fontWeight": "bold",
      },
    ]
  }
>
  Heading h3
</Text>
`;

exports[`<Heading /> renders an h4 level heading 1`] = `
<Text
  level="h4"
  style={
    Array [
      Object {
        "color": "#293241",
        "fontSize": 16,
        "fontWeight": "bold",
      },
    ]
  }
>
  Heading h4
</Text>
`;
